### PR TITLE
XLB-002 Added healthcheck scheduler for faulty routes

### PR DESCRIPTION
**Pull request by xdire**
- Adds health check worker PQ scheduler implementation as described in: https://github.com/xdire/xlb/blob/main/rfd/0001-general-design.md#health-check-scheduler
- Changes the Forwarder Dial process to loop until the good host will be selected
- Unhealthy hosts will be scheduled to health check Offer/Poll mechanics
- Per each unhealthy route, there will be one goroutine to track the state

### CODE REVIEW

File: healthcheck.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+			}
+
+		}
+	}
+}
+
+// Object to operate within queue
+type taskItem struct {
+	Id       int64
+	Value    interface{}
```
Comment (strideynet) [inline healthcheck.go:210]: Could Go Generics be leveraged here to avoid losing type safety?

File: healthcheck.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+	*pq = (*pq)[0 : n+1]
+	item := x.(*taskItem)
+	item.Index = n
+	(*pq)[n] = item
+}
+
+func (pq *taskQueue) Pop() interface{} {
+	n := len(*pq)
+	c := cap(*pq)
+	if n < (c/2) && c > 25 {
```
Comment (strideynet) [inline healthcheck.go:253]: It's difficult to understand the significance of "25" etc here without a comment explaining this behaviour.

File: healthcheck.go (hunk 1 above)
Comment (xdire) [reply healthcheck.go:210]: 100%! The only thing why this code lacks them is the developer being so long locked into a certain range of Go versions by external forces :) I assume a few next PRs (or maybe even this one) would benefit from some refactoring :)

File: healthcheck.go (hunk 2 above)
Comment (xdire) [reply healthcheck.go:253]: Certainly, the number is arbitrary and designates minimal capacity allocation, which should be fixed to a lower value, assuming we are dealing with occasional hosts going down and most cases will be lower range.

**Pull request merged**